### ISSUE TITLE
Disappearing confidence intervals

### ISSUE THREAD

**Issue by MaycocSE** [kassambara/survminer#538]
When using ggsurvplot to create kaplan meier curves, the confidence intervals are disappearing if the x axis is restricted.

Using the "lung" dataset:

```{R}
library(tidyverse)
library(survival)
library(survminer)

data <- lung
fit <- survfit(Surv(time, status) ~ sex, data)
```

The following works fine.

```{R}
ggsurvplot(fit,
           data,
           fun = "pct",
           risk.table = T,
           conf.int = T,
           break.time.by = 100,
           xlab = "Survival time (days)",
           ylab = "Survival (%)",
           xlim = c(0, 1050),
           censor.shape = "",
           ggtheme = theme_classic(),
           tables.theme = theme_cleantable(),
           axes.offset = FALSE,
           legend.title = "Sex",
           legend.labs = c("Male", "Female"),
           fontsize = 3) 
```
![KM with CI](https://user-images.githubusercontent.com/55692860/122360056-01d3f980-cf4e-11eb-979f-c38a1467159e.png)

But if I try to restrict the x-axis the confidence intervals disappear

```{R}
ggsurvplot(fit,
           data,
           fun = "pct",
           risk.table = T,
           conf.int = T,
           break.time.by = 100,
           xlab = "Survival time (days)",
           ylab = "Survival (%)",
           xlim = c(0, 800),
           censor.shape = "",
           ggtheme = theme_classic(),
           tables.theme = theme_cleantable(),
           axes.offset = FALSE,
           legend.title = "Sex",
           legend.labs = c("Male", "Female"),
           fontsize = 3) 
```
![KM no CI](https://user-images.githubusercontent.com/55692860/122360095-08fb0780-cf4e-11eb-93d7-297a06eada67.png)

This isn't an issue with conf.int.style = "step" only = "ribbon"

```{R}
ggsurvplot(fit,
           data,
           fun = "pct",
           risk.table = T,
           conf.int = T,
           conf.int.style = "step",
           break.time.by = 100,
           xlab = "Survival time (days)",
           ylab = "Survival (%)",
           xlim = c(0, 800),
           censor.shape = "",
           ggtheme = theme_classic(),
           tables.theme = theme_cleantable(),
           axes.offset = FALSE,
           legend.title = "Sex",
           legend.labs = c("Male", "Female"),
           fontsize = 3) 
```
![KM with step CI](https://user-images.githubusercontent.com/55692860/122360278-3051d480-cf4e-11eb-8d58-8a223e8b4a99.png)

````{R}
devtools::session_info()
````

R version 4.1.0 (2021-05-18)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18363)

Matrix products: default

locale:
[1] LC_COLLATE=English_United Kingdom.1252  LC_CTYPE=English_United Kingdom.1252   
[3] LC_MONETARY=English_United Kingdom.1252 LC_NUMERIC=C                           
[5] LC_TIME=English_United Kingdom.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] survminer_0.4.8 ggpubr_0.4.0    survival_3.2-11 forcats_0.5.1  
 [5] stringr_1.4.0   dplyr_1.0.6     purrr_0.3.4     readr_1.4.0    
 [9] tidyr_1.1.3     tibble_3.1.2    ggplot2_3.3.3   tidyverse_1.3.1

loaded via a namespace (and not attached):
 [1] fs_1.5.0          usethis_2.0.1     lubridate_1.7.10  devtools_2.4.2   
 [5] httr_1.4.2        rprojroot_2.0.2   tools_4.1.0       backports_1.2.1  
 [9] utf8_1.2.1        R6_2.5.0          DBI_1.1.1         colorspace_2.0-1 
[13] withr_2.4.2       prettyunits_1.1.1 processx_3.5.2    tidyselect_1.1.1 
[17] gridExtra_2.3     curl_4.3.1        compiler_4.1.0    cli_2.5.0        
[21] rvest_1.0.0       xml2_1.3.2        desc_1.3.0        labeling_0.4.2   
[25] scales_1.1.1      survMisc_0.5.5    callr_3.7.0       digest_0.6.27    
[29] foreign_0.8-81    rmarkdown_2.8     rio_0.5.26        pkgconfig_2.0.3  
[33] htmltools_0.5.1.1 sessioninfo_1.1.1 dbplyr_2.1.1      fastmap_1.1.0    
[37] rlang_0.4.11      readxl_1.3.1      rstudioapi_0.13   generics_0.1.0   
[41] farver_2.1.0      zoo_1.8-9         jsonlite_1.7.2    zip_2.2.0        
[45] car_3.0-10        magrittr_2.0.1    Matrix_1.3-3      Rcpp_1.0.6       
[49] munsell_0.5.0     fansi_0.5.0       abind_1.4-5       lifecycle_1.0.0  
[53] stringi_1.6.1     yaml_2.2.1        carData_3.0-4     pkgbuild_1.2.0   
[57] grid_4.1.0        crayon_1.4.1      lattice_0.20-44   haven_2.4.1      
[61] splines_4.1.0     hms_1.1.0         ps_1.6.0          knitr_1.33       
[65] pillar_1.6.1      ggsignif_0.6.1    pkgload_1.2.1     reprex_2.0.0     
[69] glue_1.4.2        evaluate_0.14     remotes_2.4.0     data.table_1.14.0
[73] modelr_0.1.8      vctrs_0.3.8       testthat_3.0.2    cellranger_1.1.0 
[77] gtable_0.3.0      km.ci_0.5-2       assertthat_0.2.1  cachem_1.0.5     
[81] xfun_0.23         openxlsx_4.2.3    xtable_1.8-4      broom_0.7.6      
[85] rstatix_0.7.0     memoise_2.0.0     KMsurv_0.1-5      ellipsis_0.3.2

**Comment by bjoroeKI** [comment]
I have the same issue. I need to use the full folllow-up time to see the CI's. When I cut the plot at say 20 years, the CI disappear.

**Comment by Isabellee4555** [comment]
Same issue! Help please!
This is the weird behaviour in new ggplot versions.  Any fill will disappear if data is outside the limit. 
(The way to solve this is to set the limit in geom_ribbon)

**Comment by hemonika** [comment]
I have the same issue. The problem disappears if I use an older R version (until 4.0.5), but the same package versions, and it comes back when I use R versions from 4.1.0.

**Comment by shilsenbeck** [comment]
I have the same problem. the problem disappeared when I added an explicit xlim= that extends the x-axis enough to capture the max value. For example, I am using the cancer dataset from package('survival').  After omitting some cases, time to recurrence (rtime) ranges up to 5342 days (0 nodes max recurrence=5309 days, 1-3 nodes max recurrence time=5342, >4 nodes max recurrence time=5334 days). If xlim is not specified, the confidence region for the 1-3 nodes group is omitted. Depending on the value specified, some or all confidence regions show up. Using patchwork to combine or individually display graphs brings the confidence regions back (see last graphs below). 
``` {r}
library(survival)
library(survminer)
library(Rcmdr)
library(patchwork)
data(cancer)
registry <- rotterdam
registry <- registry[registry$year>=1985,]
registry <- within(registry,{
  f.nodes <- Recode(nodes, '0="0"; 1:3="1 to 3"; 4:hi="4+"', 
                  as.factor=TRUE)
})

numSummary(registry[,"rtime", drop=FALSE], groups=registry$f.nodes, 
           statistics=c( "quantiles"), quantiles=c(0,1))

       0% 100%    n
0      36 5309 1292
1 to 3 38 5342  631
4+     45 5334  680
```

``` {r}
sfit1 <- surv_fit(Surv(rtime, recur) ~ f.nodes, data=registry)
p1 <- ggsurvplot(sfit1, data=registry, conf.int = TRUE,
                 title='Default')

p1
```
![p1](https://user-images.githubusercontent.com/13435426/136733430-4d06166a-d793-4428-8456-6f2add87b219.png)

``` {r}
p2 <- ggsurvplot(sfit1, data=registry, conf.int = TRUE,
                 title='xlim=c(0,5000)',
                 # need to specify xlim > x max to get all confints
                 xlim=c(0,5100))  
p2
```
![p2](https://user-images.githubusercontent.com/13435426/136733463-b0778461-7496-4160-a015-5afbece68f26.png)

`p1$plot+p2$plot`
![p1+p2](https://user-images.githubusercontent.com/13435426/136733478-685ad44a-3e33-4f2d-90d5-2bf7b1dac6c9.png)
`(p1$plot + theme(plot.margin = unit(c(0,20,0,0), "pt")))`
![p1+theme](https://user-images.githubusercontent.com/13435426/136733574-385fb057-67d8-45b0-84de-91ee78994ac2.png)

``` {r}
devtools::session_info()
```
- Session info ---------------------------------------------------------------
 setting  value                       
 version  R version 4.1.1 (2021-08-10)
 os       Windows 10 x64              
 system   x86_64, mingw32             
 ui       RStudio                     
 language (EN)                        
 collate  English_United States.1252  
 ctype    English_United States.1252  
 tz       America/Chicago             
 date     2021-10-10                  

- Packages -------------------------------------------------------------------
 ! package      * version  date       lib source        
   abind          1.4-5    2016-07-21 [1] CRAN (R 4.0.3)
   assertthat     0.2.1    2019-03-21 [1] CRAN (R 4.0.3)
   backports      1.2.1    2020-12-09 [1] CRAN (R 4.1.0)
   base64enc      0.1-3    2015-07-28 [1] CRAN (R 4.0.3)
   boot           1.3-28   2021-05-03 [2] CRAN (R 4.1.1)
   broom          0.7.9    2021-07-27 [1] CRAN (R 4.1.1)
   cachem         1.0.6    2021-08-19 [1] CRAN (R 4.1.1)
   callr          3.7.0    2021-04-20 [1] CRAN (R 4.1.0)
   car          * 3.0-11   2021-06-27 [1] CRAN (R 4.1.0)
   carData      * 3.0-4    2020-05-22 [1] CRAN (R 4.0.3)
   cellranger     1.1.0    2016-07-27 [1] CRAN (R 4.0.3)
   checkmate      2.0.0    2020-02-06 [1] CRAN (R 4.0.3)
   class          7.3-19   2021-05-03 [2] CRAN (R 4.1.1)
   cli            3.0.1    2021-07-17 [1] CRAN (R 4.1.1)
   cluster        2.1.2    2021-04-17 [2] CRAN (R 4.1.1)
   codetools      0.2-18   2020-11-04 [2] CRAN (R 4.1.1)
   colorspace     2.0-2    2021-06-24 [1] CRAN (R 4.1.0)
   crayon         1.4.1    2021-02-08 [1] CRAN (R 4.1.0)
   curl           4.3.2    2021-06-23 [1] CRAN (R 4.1.0)
   data.table     1.14.0   2021-02-21 [1] CRAN (R 4.1.0)
   DBI            1.1.1    2021-01-15 [1] CRAN (R 4.0.3)
   desc           1.3.0    2021-03-05 [1] CRAN (R 4.1.0)
   devtools       2.4.2    2021-06-07 [1] CRAN (R 4.1.0)
   digest         0.6.27   2020-10-24 [1] CRAN (R 4.0.3)
   dplyr          1.0.7    2021-06-18 [1] CRAN (R 4.1.0)
   e1071          1.7-8    2021-07-28 [1] CRAN (R 4.1.1)
   effects      * 4.2-0    2020-08-11 [1] CRAN (R 4.0.3)
   ellipsis       0.3.2    2021-04-29 [1] CRAN (R 4.1.0)
   fansi          0.5.0    2021-05-25 [1] CRAN (R 4.1.0)
   farver         2.1.0    2021-02-28 [1] CRAN (R 4.1.0)
   fastmap        1.1.0    2021-01-25 [1] CRAN (R 4.0.3)
   forcats        0.5.1    2021-01-27 [1] CRAN (R 4.0.3)
   foreign        0.8-81   2020-12-22 [2] CRAN (R 4.1.1)
   Formula        1.2-4    2020-10-16 [1] CRAN (R 4.0.3)
   fs             1.5.0    2020-07-31 [1] CRAN (R 4.0.3)
   generics       0.1.0    2020-10-31 [1] CRAN (R 4.0.3)
   ggplot2      * 3.3.5    2021-06-25 [1] CRAN (R 4.1.0)
   ggpubr       * 0.4.0    2020-06-27 [1] CRAN (R 4.0.5)
   ggsignif       0.6.3    2021-09-09 [1] CRAN (R 4.1.1)
   glue           1.4.2    2020-08-27 [1] CRAN (R 4.1.0)
   gridExtra      2.3      2017-09-09 [1] CRAN (R 4.0.3)
   gtable         0.3.0    2019-03-25 [1] CRAN (R 4.0.3)
   haven          2.4.3    2021-08-04 [1] CRAN (R 4.1.1)
   Hmisc          4.5-0    2021-02-28 [1] CRAN (R 4.1.0)
   hms            1.1.0    2021-05-17 [1] CRAN (R 4.1.0)
   htmlTable      2.2.1    2021-05-18 [1] CRAN (R 4.1.0)
   htmltools      0.5.2    2021-08-25 [1] CRAN (R 4.1.1)
   htmlwidgets    1.5.4    2021-09-08 [1] CRAN (R 4.1.1)
   insight        0.14.4   2021-09-02 [1] CRAN (R 4.1.1)
   jpeg           0.1-9    2021-07-24 [1] CRAN (R 4.1.1)
   km.ci          0.5-2    2009-08-30 [1] CRAN (R 4.0.5)
   KMsurv         0.1-5    2012-12-03 [1] CRAN (R 4.0.3)
   knitr          1.34     2021-09-09 [1] CRAN (R 4.1.1)
   labeling       0.4.2    2020-10-20 [1] CRAN (R 4.0.3)
   lattice        0.20-44  2021-05-02 [2] CRAN (R 4.1.1)
   latticeExtra   0.6-29   2019-12-19 [1] CRAN (R 4.0.3)
   lifecycle      1.0.0    2021-02-15 [1] CRAN (R 4.1.0)
   lme4           1.1-27.1 2021-06-22 [1] CRAN (R 4.1.0)
   magrittr       2.0.1    2020-11-17 [1] CRAN (R 4.1.0)
   MASS           7.3-54   2021-05-03 [2] CRAN (R 4.1.1)
   Matrix         1.3-4    2021-06-01 [2] CRAN (R 4.1.1)
   memoise        2.0.0    2021-01-26 [1] CRAN (R 4.1.0)
   minqa          1.2.4    2014-10-09 [1] CRAN (R 4.0.3)
   mitools        2.4      2019-04-26 [1] CRAN (R 4.0.3)
   multcomp       1.4-17   2021-04-29 [1] CRAN (R 4.1.0)
   munsell        0.5.0    2018-06-12 [1] CRAN (R 4.0.3)
   mvtnorm        1.1-2    2021-06-07 [1] CRAN (R 4.1.0)
   nlme           3.1-152  2021-02-04 [2] CRAN (R 4.1.1)
   nloptr         1.2.2.2  2020-07-02 [1] CRAN (R 4.0.3)
   nnet           7.3-16   2021-05-03 [2] CRAN (R 4.1.1)
   nortest        1.0-4    2015-07-30 [1] CRAN (R 4.0.3)
   openxlsx       4.2.4    2021-06-16 [1] CRAN (R 4.1.0)
   patchwork    * 1.1.1    2020-12-17 [1] CRAN (R 4.0.4)
   pillar         1.6.2    2021-07-29 [1] CRAN (R 4.1.1)
   pkgbuild       1.2.0    2020-12-15 [1] CRAN (R 4.0.3)
   pkgconfig      2.0.3    2019-09-22 [1] CRAN (R 4.0.3)
   pkgload        1.2.2    2021-09-11 [1] CRAN (R 4.1.1)
   png            0.1-7    2013-12-03 [1] CRAN (R 4.0.3)
   prettyunits    1.1.1    2020-01-24 [1] CRAN (R 4.0.3)
   processx       3.5.2    2021-04-30 [1] CRAN (R 4.1.0)
   proxy          0.4-26   2021-06-07 [1] CRAN (R 4.1.0)
   ps             1.6.0    2021-02-28 [1] CRAN (R 4.1.0)
   purrr          0.3.4    2020-04-17 [1] CRAN (R 4.0.3)
   R6             2.5.1    2021-08-19 [1] CRAN (R 4.1.1)
   Rcmdr        * 2.7-1    2020-10-07 [1] CRAN (R 4.1.0)
   RcmdrMisc    * 2.7-1    2020-08-13 [1] CRAN (R 4.1.0)
   RColorBrewer   1.1-2    2014-12-07 [1] CRAN (R 4.0.3)
   Rcpp           1.0.7    2021-07-07 [1] CRAN (R 4.1.0)
   readxl         1.3.1    2019-03-13 [1] CRAN (R 4.0.3)
   relimp         1.0-5    2016-03-30 [1] CRAN (R 4.0.3)
   remotes        2.4.0    2021-06-02 [1] CRAN (R 4.1.0)
   rio            0.5.27   2021-06-21 [1] CRAN (R 4.1.0)
 D rJava        * 1.0-4    2021-04-29 [1] CRAN (R 4.1.0)
   rlang          0.4.11   2021-04-30 [1] CRAN (R 4.1.0)
   rpart          4.1-15   2019-04-12 [2] CRAN (R 4.1.1)
   rprojroot      2.0.2    2020-11-15 [1] CRAN (R 4.0.3)
   rstatix        0.7.0    2021-02-13 [1] CRAN (R 4.0.5)
   rstudioapi     0.13     2020-11-12 [1] CRAN (R 4.0.3)
   sandwich     * 3.0-1    2021-05-18 [1] CRAN (R 4.1.0)
   scales         1.1.1    2020-05-11 [1] CRAN (R 4.0.3)
   sessioninfo    1.1.1    2018-11-05 [1] CRAN (R 4.1.0)
   stringi        1.7.4    2021-08-25 [1] CRAN (R 4.1.1)
   stringr        1.4.0    2019-02-10 [1] CRAN (R 4.1.0)
   survey         4.1-1    2021-07-19 [1] CRAN (R 4.1.1)
   survival     * 3.2-11   2021-04-26 [2] CRAN (R 4.1.1)
   survminer    * 0.4.9    2021-03-09 [1] CRAN (R 4.0.5)
   survMisc       0.5.5    2018-07-05 [1] CRAN (R 4.0.5)
   tcltk2         1.2-11   2014-12-20 [1] CRAN (R 4.0.3)
   testthat       3.0.4    2021-07-01 [1] CRAN (R 4.1.0)
   TH.data        1.0-10   2019-01-21 [1] CRAN (R 4.0.3)
   tibble         3.1.4    2021-08-25 [1] CRAN (R 4.1.1)
   tidyr          1.1.3    2021-03-03 [1] CRAN (R 4.1.0)
   tidyselect     1.1.1    2021-04-30 [1] CRAN (R 4.1.0)
   usethis        2.0.1    2021-02-10 [1] CRAN (R 4.1.0)
   utf8           1.2.2    2021-07-24 [1] CRAN (R 4.1.1)
   vctrs          0.3.8    2021-04-29 [1] CRAN (R 4.1.0)
   withr          2.4.2    2021-04-18 [1] CRAN (R 4.1.0)
   xfun           0.26     2021-09-14 [1] CRAN (R 4.1.1)
   xtable         1.8-4    2019-04-21 [1] CRAN (R 4.0.3)
   zip            2.2.0    2021-05-31 [1] CRAN (R 4.1.0)
   zoo            1.8-9    2021-03-09 [1] CRAN (R 4.1.0)

[1] C:/Users/susan/OneDrive/Documents/R/win-library/4.1
[2] C:/Program Files/R/R-4.1.1/library

 D -- DLL MD5 mismatch, broken installation.

**Comment by shilsenbeck** [comment]
I think mhmdrahouma's solution will work, EXCEPT, I think that DX_LASTCONTACT_DEATH_MONTHS_R1 needs to be truncated to 72, not 79, AND the STATUS needs to be set to not "Dead" for anyone that was truncated in order to preserve the same KM estimates up to 72 months.

**Comment by stephan1312** [comment]
> Same issue! Help please! This is the weird behaviour in new ggplot versions. Any fill will disappear if data is outside the limit. (The way to solve this is to set the limit in geom_ribbon)

I can confirm that it depends on the **ggplot2** version. When I manually downgraded to **ggplot2 version 3.2.1** (the oldest still working with the current tidyverse package) and downgraded survminer to 0.4.8 (because survminer 0.4.9 requires a current ggplot2), the confidence intervals re-appeared.

**Comment by jmanitz** [comment]
I have the same issue here. Has it been escalated to the ggplot's github already? 
For survminer, it seems that even of xlim is not specified that the x axis is cut. I would suggest to implement the default xlim = c(0, max(fit$time))

**Comment by ljacks-stats** [comment]
Experiencing the same issue in R 4.1.1, using xlim = c(0, max(fit$time)) solves the issue.

**Comment by MaycocSE** [comment]
The issue seems to be fixed using R 4.2.0 and survival 3.3-1
